### PR TITLE
Update lisk-genesis to support on-chain architecture #5640

### LINKED ISSUE #5640: Update lisk-genesis to support on-chain architecture
### Description

- validation happens at the module and chain, so it does not need to be done in lisk-genesis.
- lisk-genesis should accept the account schemas defined in the modules, and create use the schema

### Motivation

- Account schemas are defined in the modules, so lisk-genesis should not depends on it internally. It should obtain from the input.

### Acceptance Criteria
- Specific account properties is not defined
- Build and test for lisk-genesis is passing

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5640 

### How was it solved?

- Remove validation logic since it is written in the module.
- Update create function input to expect accountSchema defined in modules
- Update to use schema defined in lisk-chain

### How was it tested?

- Update all the tests
- Use sample schema which is same as in module to test the basic feature

### CODE REVIEW

File: elements/lisk-chain/src/utils/account.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	for (const [name, schema] of Object.entries(accountSchemas)) {
+		const { default: defaultValue, ...schemaWithoutDefault } = schema;
+		accountSchema.properties[name] = schemaWithoutDefault;
+		(accountSchema.required as string[]).push(name);
+		defaultAccount[name] = defaultValue;
+	}
+	return {
+		schema: accountSchema,
+		defaultAccount,
+	};
```
Comment (nazarhussain) [inline elements/lisk-chain/src/utils/account.ts:34]: Suggest to use the pattern we are using on other places. `default` should be the key under schema object containing default account.

File: elements/lisk-chain/src/chain.ts [hunk 2]
```diff
@@ -124,14 +123,9 @@ export class Chain {
...
 
-		this._accountSchema = objects.cloneDeep(baseAccountSchema);
-		this._defaultAccount = {};
-		for (const [name, schema] of Object.entries(accounts)) {
-			const { default: defaultValue, ...schemaWithoutDefault } = schema;
-			this._accountSchema.properties[name] = schemaWithoutDefault;
-			(this._accountSchema.required as string[]).push(name);
-			this._defaultAccount[name] = defaultValue;
-		}
+		const { defaultAccount, schema } = getAccountSchemaAndDefault(accounts);
```
Comment (nazarhussain) [inline elements/lisk-chain/src/chain.ts:126]: This argument here looks like accounts objects while the function accepts the account schema. This is related to earlier comment on other PR to rename `Chain` constructor arguments name `accounts` to `accountSchema`

File: elements/lisk-chain/src/utils/account.ts (hunk 1 above)
Comment (nazarhussain) [inline elements/lisk-chain/src/utils/account.ts:34]: Should not we use the same pattern we use on other places. `default` should be a key under schema object, containing default account.

File: elements/lisk-chain/src/chain.ts (hunk 2 above)
Comment (shuse2) [reply elements/lisk-chain/src/chain.ts:126]: For the constructor param I added a point on https://github.com/LiskHQ/lisk-sdk/issues/5639

**Pull request merged**